node/pkg/common/chainlock.go

```diff
@@ -48,7 +51,12 @@ const minMsgLength = 88 // Marshalled length with empty payload
 func (msg *MessagePublication) Marshal() ([]byte, error) {
 	buf := new(bytes.Buffer)
 
-	buf.Write(msg.TxHash[:])
+	if len(msg.TxID) > math.MaxUint32 {
```

Comment:
Can we make this much more aggressive? The maximum permissible value here would mean a 4GB byte array. We could either cap at `MaxUint8` or at `64` explicitly?